egs/swbd/s5c/local/reverberate/reverberate_data_dir.py

```diff
@@ -0,0 +1,165 @@
+
+class list_cyclic_iterator:
+  def __init__(self, list, random_seed = 0):
+    self.list_index = 0
+    self.list = list
+    random.seed(random_seed)
+    random.shuffle(self.list)
+
+  def next(self):
+    item = self.list[self.list_index]
+    self.list_index = (self.list_index + 1) % len(self.list)
+    return item
+
+def return_nonempty_lines(lines):
+  new_lines = []
+  for line in lines:
+    if len(line.strip()) > 0:
+      new_lines.append(line.strip())
+
+  return new_lines
+
+
+def GetArgs():
+    # we add compulsary arguments as named arguments for readability
+    parser = argparse.ArgumentParser(description="Writes config files and variables "
+                                                 "for TDNNs creation and training",
+                                     epilog="See steps/nnet3/tdnn/train.sh for example.")
+
+    parser.add_argument("--rir-list", type=str, default = None,
+                        help="RIR information file")
+    parser.add_argument("--noise-list", type=str, default = None,
+                        help="Noise information file")
+    parser.add_argument("--num-replications", type=int, dest = "num_replica", default = 1,
+                        help="Number of replicate to generated for the data")
+    parser.add_argument("input_dir",
+                        help="Input data directory")
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if args.rir_list is None:
+        raise Exception("Rir information file must be provided")
+    
+    if not os.path.isfile(args.rir_list):
+        raise Exception(args.rir_list + "not found")
+    
+    if args.noise_list is not None:
+        if not os.path.isfile(args.noise_list):
+            raise Exception(args.noise_list + "not found")
+
+    return args
+
+
+def Corrupt_wav(input_dir, output_dir, rir_list, num_replica, prefix):
+    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
+    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
+    command_list = []
+    for i in range(num_replica):
+        for j in range(len(wav_files)):
+            split1 = wav_files[j].split()
+            wav_pipe = " ".join(split1[1:])
+            output_wav_id = prefix + str(i) + "_" + split1[0]
+            rir = rirs.next().split()
+            rir_file_location = rir[-1]
+            command_list.append("{2} {0} wav-reverberate - {1} - |\n".format(wav_pipe, rir_file_location, output_wav_id))
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+def Corrupt_wav_noise(input_dir, output_dir, rir_list, noise_list, snr_array, num_replica, prefix):
```

Comment:
User should be able to specify an option which specifies whether point sources have to be added. Also @vimalmanohar requires foreground and background noises, it might be better if @vimalmanohar decides this function signature.